src/hostkey.c

```diff
@@ -308,20 +306,16 @@ hostkey_method_ssh_dss_init(LIBSSH2_SESSION * session,
     if(_libssh2_match_string(&buf, "ssh-dss") != 0)
         return -1;
 
-    p_len = _libssh2_get_c_string(&buf, &p);
-    if(p_len < 0)
+    if(_libssh2_get_string(&buf, &p, &p_len) != 0)
```

Comment:
personally, I'd prefer these lines without the `!= 0` part. Like this:
~~~c
if(_libssh2_get_string(&buf, &p, &p_len))
    return -1;

Reply:
I'm fine with that and can make these changes. I typically use explicit value checks to ensure the caller knows what the return types are.